examples/shapes/src/main.rs

```diff
@@ -0,0 +1,194 @@
+use anyhow::Result;
+use std::num::NonZeroUsize;
+use std::sync::Arc;
+use vello::kurbo::{Circle, Ellipse, Line, RoundedRect, Stroke};
+use vello::peniko::Color;
+use vello::util::RenderSurface;
+use vello::RendererOptions;
+use vello::{kurbo::Affine, util::RenderContext, AaConfig, Renderer, Scene};
+use winit::{dpi::LogicalSize, window::WindowBuilder};
+use winit::{event::*, event_loop::ControlFlow};
+use winit::{event_loop::EventLoop, window::Window};
+
+// Simple struct to hold the state of the renderer
+pub struct RenderState<'s> {
+    // The fields MUST be in this order, so that the surface is dropped before the window
+    surface: RenderSurface<'s>,
+    window: Arc<Window>,
+}
+
+// Helper function that creates a Winit window and returns it (wrapped in an Arc for thread safety)
+fn create_window(event_loop: &winit::event_loop::EventLoopWindowTarget<()>) -> Arc<Window> {
+    Arc::new(
+        WindowBuilder::new()
+            .with_inner_size(LogicalSize::new(1044, 800))
+            .with_resizable(true)
+            .with_title("Vello Shapes")
+            .build(event_loop)
+            .unwrap(),
+    )
+}
+
+// Runs the Winit event loop and renders the shapes
+fn run(event_loop: EventLoop<()>, mut render_cx: RenderContext) {
+    // Create an empty vec of `Renderer`s.
+    let mut renderers: Vec<Option<Renderer>> = vec![];
+    // Create a RenderState wrapped in an Option.
+    let mut render_state = None::<RenderState>;
+    // Cache a window so that it can be reused when the app is resumed after being suspended
+    let mut cached_window = None;
+    // Create a new scene
+    let mut scene = Scene::new();
+
+    // Run the event loop
+    event_loop
+        .run(move |event, event_loop| match event {
```

Comment:
Four of these five comments don't give any information which can't be trivially inferred from the line of code they're attached to.
I think these are likely to add comment fatigue, meaning that users may skip reading the actually useful comments later

Reply:
The comment you've left wasn't the one which is informative.
```rust
// Cache a window so that it can be reused when the app is resumed after being suspended
```
is a useful comment - I don't think the others added anything though